<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<title>Basic usage • ember</title>
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><link href="../deps/font-awesome-6.4.2/css/all.min.css" rel="stylesheet">
<link href="../deps/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet">
<script src="../deps/headroom-0.11.0/headroom.min.js"></script><script src="../deps/headroom-0.11.0/jQuery.headroom.min.js"></script><script src="../deps/bootstrap-toc-1.0.1/bootstrap-toc.min.js"></script><script src="../deps/clipboard.js-2.0.11/clipboard.min.js"></script><script src="../deps/search-1.0.0/autocomplete.jquery.min.js"></script><script src="../deps/search-1.0.0/fuse.min.js"></script><script src="../deps/search-1.0.0/mark.min.js"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Basic usage">
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>


    <nav class="navbar navbar-expand-lg fixed-top bg-light" data-bs-theme="light" aria-label="Site navigation"><div class="container">

    <a class="navbar-brand me-2" href="../index.html">ember</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">0.1.0</small>


    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item"><a class="nav-link" href="../reference/index.html">Reference</a></li>
<li class="active nav-item dropdown">
  <button class="nav-link dropdown-toggle" type="button" id="dropdown-articles" data-bs-toggle="dropdown" aria-expanded="false" aria-haspopup="true">Articles</button>
  <ul class="dropdown-menu" aria-labelledby="dropdown-articles">
<li><a class="dropdown-item" href="../articles/basic-use.html">Basic usage</a></li>
  </ul>
</li>
      </ul>
<ul class="navbar-nav">
<li class="nav-item"><form class="form-inline" role="search">
 <input class="form-control" type="search" name="search-input" id="search-input" autocomplete="off" aria-label="Search site" placeholder="Search for" data-search-index="../search.json">
</form></li>
<li class="nav-item"><a class="external-link nav-link" href="https://github.com/chronchi/ember/" aria-label="GitHub"><span class="fa fab fa-github fa-lg"></span></a></li>
      </ul>
</div>


  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">

      <h1>Basic usage</h1>
            
      
      <small class="dont-index">Source: <a href="https://github.com/chronchi/ember/blob/HEAD/vignettes/basic-use.Rmd" class="external-link"><code>vignettes/basic-use.Rmd</code></a></small>
      <div class="d-none name"><code>basic-use.Rmd</code></div>
    </div>

    
    
<p>In this tutorial we explain shortly how to use the most important
features of ember by analysing a real world dataset, the ABiM 100
dataset<a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content='&lt;p&gt;&lt;a href="https://data.mendeley.com/datasets/yzxtxn4nmd/3" class="external-link uri"&gt;https://data.mendeley.com/datasets/yzxtxn4nmd/3&lt;/a&gt;&lt;/p&gt;'><sup>1</sup></a>.
This is a Swedish cohort of early stage breast cancer with a total of
100 samples. What makes it special is that the consortium did RNA-seq
and also sent samples to prosigna to get the PAM50 intrinsic molecular
subtype and also the risk of recurrence (ROR).</p>
<p>To install the package run R and the following lines:</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw">if</span> <span class="op">(</span><span class="op">!</span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">require</a></span><span class="op">(</span><span class="st"><a href="https://devtools.r-lib.org/" class="external-link">"devtools"</a></span>, quietly <span class="op">=</span> <span class="cn">TRUE</span><span class="op">)</span><span class="op">)</span></span>
<span>    <span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"chronchi/ember"</span><span class="op">)</span></span></code></pre></div>
<div class="section level2">
<h2 id="packages-for-tutorial">Packages for tutorial<a class="anchor" aria-label="anchor" href="#packages-for-tutorial"></a>
</h2>
<p>In this tutorial we will use several packages besides
<code>ember</code>. These packages should be available after installing
<code>ember</code>.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://chronchi.github.io/ember">ember</a></span><span class="op">)</span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://bioconductor.org/packages/SummarizedExperiment" class="external-link">SummarizedExperiment</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://igordot.github.io/msigdbr/" class="external-link">msigdbr</a></span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level2">
<h2 id="get-pca-coordinates-and-plot">Get PCA coordinates and plot<a class="anchor" aria-label="anchor" href="#get-pca-coordinates-and-plot"></a>
</h2>
<p>The first step of <code>ember</code> is to get the samples embedded
in the common space of all molecular samples. Here we can plot the ABiM
samples on top of all available all samples from TCGA, METABRIC and
SCAN-B as of November 2023. First we need to normalize the data by using
the function <code>get_final_raking_values</code>. The embedding is
obtained by using the “small” version of embedding using the 1044 genes.
The method is robust for missing genes, so if not all 1044 genes are
available you can still use it. Given your dataset the only arguments
one needs to change then is the <code>sum_exp</code> and
<code>assay_to_use</code> arguments. The <code>sum_exp</code>
corresponds to your dataset as a SummarizedExperiment object and the
<code>assay_to_use</code> corresponds to the slot in the
SummarizedExperiment object in which the normalized data is saved. The
package internally uses <code>assay(sum_exp, assay_to_use)</code> to
fetch the data. See the SummarizedExperiment package for more
information on this data type<a class="footnote-ref" tabindex="0" data-bs-toggle="popover" data-bs-content='&lt;p&gt;&lt;a href="https://bioconductor.org/packages/release/bioc/vignettes/SummarizedExperiment/inst/doc/SummarizedExperiment.html" class="external-link uri"&gt;https://bioconductor.org/packages/release/bioc/vignettes/SummarizedExperiment/inst/doc/SummarizedExperiment.html&lt;/a&gt;&lt;/p&gt;'><sup>2</sup></a>.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># first calculate the normalization. by default it uses the </span></span>
<span><span class="co"># small PCA model</span></span>
<span><span class="va">abim_100_normalized</span> <span class="op">&lt;-</span> <span class="fu">ember</span><span class="fu">::</span><span class="fu"><a href="../reference/get_final_ranking_values.html">get_final_ranking_values</a></span><span class="op">(</span></span>
<span>    sum_exp <span class="op">=</span> <span class="va">abim_100</span>, assay_to_use <span class="op">=</span> <span class="st">"log2fpkm"</span>, </span>
<span>    most_variable_genes <span class="op">=</span> <span class="st">"small"</span></span>
<span><span class="op">)</span></span>
<span><span class="co">#&gt; Total number of stable genes: 44</span></span>
<span><span class="co">#&gt; Total number of genes: 1044</span></span>
<span><span class="co">#&gt; Number of samples: 100</span></span>
<span></span>
<span><span class="co"># calculate the embedding for the abim</span></span>
<span><span class="va">abim_100_normalized</span> <span class="op">&lt;-</span> <span class="fu">ember</span><span class="fu">::</span><span class="fu"><a href="../reference/get_pca_coordinates.html">get_pca_coordinates</a></span><span class="op">(</span></span>
<span>    <span class="va">abim_100_normalized</span>, which_pca <span class="op">=</span> <span class="st">"small"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span></span>
<span><span class="co"># get now the base plot so we can overlay the samples into the </span></span>
<span><span class="co"># embedding</span></span>
<span><span class="va">base_plot</span> <span class="op">&lt;-</span> <span class="fu">ember</span><span class="fu">::</span><span class="fu"><a href="../reference/get_base_plot.html">get_base_plot</a></span><span class="op">(</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># plot on top of the base plot to have an idea where the samples</span></span>
<span><span class="va">p</span> <span class="op">&lt;-</span> <span class="va">base_plot</span> <span class="op">+</span> </span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span></span>
<span>        <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">abim_100_normalized</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>            <span class="va">data.frame</span>,</span>
<span>        mapping <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">PC3</span>, y <span class="op">=</span> <span class="va">PC4</span>, color <span class="op">=</span> <span class="va">pam50</span><span class="op">)</span>,</span>
<span>        size <span class="op">=</span> <span class="fl">5</span>,</span>
<span>        alpha <span class="op">=</span> <span class="fl">1</span>,</span>
<span>        shape <span class="op">=</span> <span class="fl">17</span></span>
<span>    <span class="op">)</span></span></code></pre></div>
<p>The plot below shows how the ABIM samples are in the common space.
They are all over the place as expected. We color code by the PAM50
obtained from Prosigna.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">p</span></span></code></pre></div>
<p><img src="basic-use_files/figure-html/unnamed-chunk-4-1.png" width="700"></p>
<p>We can see how close they are to the other cohorts in the space of
batch effects.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">base_plot_batch</span> <span class="op">&lt;-</span> <span class="fu">ember</span><span class="fu">::</span><span class="fu"><a href="../reference/get_base_plot.html">get_base_plot</a></span><span class="op">(</span></span>
<span>    x <span class="op">=</span> <span class="st">"PC1"</span>, </span>
<span>    y <span class="op">=</span> <span class="st">"PC2"</span>,</span>
<span>    color <span class="op">=</span> <span class="st">"cohort"</span></span>
<span><span class="op">)</span></span>
<span></span>
<span><span class="co"># plot on top of the base plot to have an idea where the samples</span></span>
<span><span class="va">p_batch</span> <span class="op">&lt;-</span> <span class="va">base_plot_batch</span> <span class="op">+</span> </span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span></span>
<span>        <span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">abim_100_normalized</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>            <span class="va">data.frame</span>,</span>
<span>        mapping <span class="op">=</span> <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">PC1</span>, y <span class="op">=</span> <span class="va">PC2</span><span class="op">)</span>,</span>
<span>        color <span class="op">=</span> <span class="st">"black"</span>,</span>
<span>        size <span class="op">=</span> <span class="fl">3</span>,</span>
<span>        alpha <span class="op">=</span> <span class="fl">1</span>,</span>
<span>        shape <span class="op">=</span> <span class="fl">17</span></span>
<span>    <span class="op">)</span></span>
<span></span>
<span><span class="va">p_batch</span></span></code></pre></div>
<p><img src="basic-use_files/figure-html/unnamed-chunk-5-1.png" width="700"></p>
<p>Makes sense since ABiM is a Swedish cohort that was obtained in a
similar way to SCAN-B.</p>
</div>
<div class="section level2">
<h2 id="regress-data-and-calculate-scores">Regress data and calculate scores<a class="anchor" aria-label="anchor" href="#regress-data-and-calculate-scores"></a>
</h2>
<p>In this section we can explain how we can <code>ember</code> to
calculate scores for samples individually by first calculating their
position in a common space with more genes (8253) and then regressing
the batch effects from the first two components. In this way we can
compare scores across different cohorts even.</p>
<p>We now use the two functions <code>get_regressed_data</code> to first
calculate the normalized and regressed dataset containing in the end
8253 genes (if all of them are available) and then followed by
<code>get_scores_regressed</code> to calculate the scores for all the 50
Hallmark pathways from the Molecular Signature Database (MSigDB). The
function <code>get_gene_sets</code> is using the <code>msigdbr</code>
function under the hood to fetch the datasets and prepares the object to
be used by <code>get_scores_regressed</code>.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># we first get the gene sets to calculate the scores</span></span>
<span><span class="va">gene_sets</span> <span class="op">&lt;-</span> <span class="fu">ember</span><span class="fu">::</span><span class="fu"><a href="../reference/get_gene_sets.html">get_gene_sets</a></span><span class="op">(</span>species <span class="op">=</span> <span class="st">"Homo sapiens"</span>, category <span class="op">=</span> <span class="st">"H"</span><span class="op">)</span></span>
<span></span>
<span><span class="va">abim_100_regressed</span> <span class="op">&lt;-</span> <span class="fu">ember</span><span class="fu">::</span><span class="fu"><a href="../reference/get_regressed_data.html">get_regressed_data</a></span><span class="op">(</span></span>
<span>    sum_exp <span class="op">=</span> <span class="va">abim_100</span>, assay_to_use <span class="op">=</span> <span class="st">"log2fpkm"</span></span>
<span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="co"># calculate the scores for the regressed data</span></span>
<span>    <span class="fu">ember</span><span class="fu">::</span><span class="fu"><a href="../reference/get_scores_regressed.html">get_scores_regressed</a></span><span class="op">(</span>sum_exp <span class="op">=</span> <span class="va">.</span>, gene_sets <span class="op">=</span> <span class="va">gene_sets</span><span class="op">)</span></span>
<span><span class="co">#&gt; Total number of stable genes: 44</span></span>
<span><span class="co">#&gt; Total number of genes: 8253</span></span>
<span><span class="co">#&gt; Number of samples: 100</span></span>
<span><span class="co">#&gt; [1] "Normalization done."</span></span></code></pre></div>
<p>The functions <code>get_regressed_data</code> and
<code>get_scores_regressed</code> return SummarizedExperiment objects,
so you can use the usual functions from the SummarizedExperiment package
to fetch the metadata and the normalized regressed data (saved in the
slot <code>regressed</code>). There are 50 new columns with the names
“regressed_” and the name of the pathway in the metadata.</p>
<p>Moreover, one can also check the regressed gene expression levels by
checking the assay “regressed”.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">genes_of_interest</span> <span class="op">&lt;-</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"ESR1"</span>, <span class="st">"TFF1"</span><span class="op">)</span></span>
<span></span>
<span><span class="fu">SummarizedExperiment</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">assay</a></span><span class="op">(</span><span class="va">abim_100_regressed</span>, <span class="st">"regressed"</span><span class="op">)</span><span class="op">[</span></span>
<span>    <span class="va">genes_of_interest</span>, </span>
<span><span class="op">]</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>    <span class="va">t</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="va">data.frame</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span>x <span class="op">=</span> <span class="va">ESR1</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span></span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_histogram.html" class="external-link">geom_histogram</a></span><span class="op">(</span>bins <span class="op">=</span> <span class="fl">20</span><span class="op">)</span> <span class="op">+</span> </span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span></span>
<span>        y <span class="op">=</span> <span class="st">"Count"</span>, </span>
<span>        title <span class="op">=</span> <span class="st">"Distribution of ESR1\nregressed expression levels"</span></span>
<span>    <span class="op">)</span> <span class="op">+</span> </span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span>base_size <span class="op">=</span> <span class="fl">20</span><span class="op">)</span></span></code></pre></div>
<p><img src="basic-use_files/figure-html/unnamed-chunk-7-1.png" width="700"></p>
<p>This is a bimodal distribution as expected. Lower levels are
associated with the Estrogen Receptor (ER) negative tumors and high
levels with ER+.</p>
</div>
<div class="section level2">
<h2 id="using-the-regressed-scores">Using the regressed scores<a class="anchor" aria-label="anchor" href="#using-the-regressed-scores"></a>
</h2>
<p>We can now compare the different conditions based on the pathway
scores obtained previously. As a validation step, we compare the
estrogen receptor signaling signature (HALLMARK_ESTROGEN_RESPONSE_EARLY)
between ER IHC positive and negative tumors. We expect to see big
differences between these two subtypes.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/pkg/SummarizedExperiment/man/SummarizedExperiment-class.html" class="external-link">colData</a></span><span class="op">(</span><span class="va">abim_100_regressed</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="va">data.frame</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu">dplyr</span><span class="fu">::</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">ER</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span></span>
<span>        x <span class="op">=</span> <span class="va">ER</span>, </span>
<span>        y <span class="op">=</span> <span class="va">regressed_HALLMARK_ESTROGEN_RESPONSE_EARLY</span>,</span>
<span>        color <span class="op">=</span> <span class="va">ER</span></span>
<span>    <span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_jitter.html" class="external-link">geom_jitter</a></span><span class="op">(</span>show.legend <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">+</span> </span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_boxplot.html" class="external-link">geom_boxplot</a></span><span class="op">(</span>alpha <span class="op">=</span> <span class="fl">0.5</span>, outlier.shape <span class="op">=</span> <span class="cn">NA</span>, show.legend <span class="op">=</span> <span class="cn">FALSE</span><span class="op">)</span> <span class="op">+</span> </span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span></span>
<span>        x <span class="op">=</span> <span class="st">"ER IHC"</span>, </span>
<span>        y <span class="op">=</span> <span class="st">"Estrogen early"</span>,</span>
<span>        title <span class="op">=</span> <span class="st">"Scores calculated on\nregressed data"</span></span>
<span>    <span class="op">)</span> <span class="op">+</span> </span>
<span>    <span class="fu">ggplot2</span><span class="fu">::</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggtheme.html" class="external-link">theme_bw</a></span><span class="op">(</span>base_size <span class="op">=</span> <span class="fl">15</span><span class="op">)</span></span></code></pre></div>
<p><img src="basic-use_files/figure-html/unnamed-chunk-8-1.png" width="480"></p>
<p>We are able to see the distinctions as expected.</p>
</div>

  </main><aside class="col-md-3"><nav id="toc" aria-label="Table of contents"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p>Developed by Carlos Ronchi.</p>
</div>

<div class="pkgdown-footer-right">
  <p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.1.0.</p>
</div>

    </footer>
</div>





  </body>
</html>
